File: src/main/java/leets/leenk/domain/birthday/application/dto/response/UpcomingBirthdayUserResponse.java
package leets.leenk.domain.birthday.application.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonUnwrapped;
import io.swagger.v3.oas.annotations.media.Schema;
import leets.leenk.domain.user.application.dto.response.UserProfileResponse;
import lombok.Builder;

import java.time.LocalDate;

@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public record UpcomingBirthdayUserResponse(
        @JsonUnwrapped
        @Schema(implementation = UserProfileResponse.class)
        UserProfileResponse profile,

        @Schema(description = "사용자 생일", example = "2001-06-15")
        LocalDate birthday
) {
}

File: src/main/kotlin/leets/leenk/domain/birthday/application/dto/response/UpcomingBirthdayUserResponse.kt
package leets.leenk.domain.birthday.application.dto.response

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonUnwrapped
import io.swagger.v3.oas.annotations.media.Schema
import leets.leenk.domain.user.application.dto.response.UserProfileResponse
import java.time.LocalDate

@JsonInclude(JsonInclude.Include.NON_NULL)
data class UpcomingBirthdayUserResponse(
    @field:JsonUnwrapped
    @field:Schema(implementation = UserProfileResponse::class)
    val profile: UserProfileResponse,
    @field:Schema(description = "사용자 생일", example = "2001-06-15")
    val birthday: LocalDate,
)
